db/migrate/20241017013023_reencrypt_password_scramsha.rb

```diff
@@ -0,0 +1,17 @@
+class ReencryptPasswordScramsha < ActiveRecord::Migration[6.1]
+  def up
+    say_with_time('Reencrypting database user password with scram-sha-256') do
+      database_yml = YAML.safe_load(Rails.root.join("config", "database.yml"))
+      username = database_yml[Rails.env]["username"]
+      password = MiqPassword.try_decrypt(database_yml[Rails.env]["password"])
+
+      execute <<-SQL
+        SET password_encryption='scram-sha-256';
+      SQL
+
+      execute <<-SQL
+        ALTER ROLE #{username} WITH PASSWORD '#{password}';
```

Comment:
As discussed, let's not send the bare password, but instead pre-encrypt it.

```
password = connection.raw_connection.encrypt_password(username, password, "scram-sha-256")
```

Reply:
Done